pandas/tests/scalar/timestamp/methods/test_replace.py

```diff
@@ -192,3 +192,23 @@ def test_replace_preserves_fold(self, fold):
         ts_replaced = ts.replace(second=1)
 
         assert ts_replaced.fold == fold
+
+    def test_replace_unit(self):
+        # GH#57749
+        ts = Timestamp("2023-07-15 23:08:12")
+        ts1 = Timestamp("2023-07-15 23:08:12.134567")
+        ts2 = Timestamp("2023-07-15 23:08:12.134567123")
+        ts = ts.replace(microsecond=999)
+        assert ts.unit == "ms"
+        ts = ts.replace(microsecond=ts1.microsecond)
+        assert ts.unit == "us"
+        assert ts == ts1
+        ts = ts.replace(nanosecond=ts2.nanosecond)
+        assert ts.unit == "ns"
+        assert ts == ts2
+
+    def test_replace_resets_to_more_precise(self):
+        # GH#57749
+        ts = Timestamp(year=2020, month=1, day=1, nanosecond=5)
+        result = ts.replace(nanosecond=0)
```

Comment:
Could you add test cases where a similar zeroing results in a resolution of `"ms"` and `"us"`?